From d703e969fac07c10122beb8d907b9f918c5fd98b Mon Sep 17 00:00:00 2001
From: Andrew Burgess <andrew.burgess@embecosm.com>
Date: Tue, 6 Mar 2018 13:20:14 +0000
Subject: [PATCH 4/9] gdb/riscv: Fix type when reading register from regcache

Should use a ULONGEST when reading from the regcache.

gdb/ChangeLog:

	* riscv-tdep.c (riscv_return_value): Change type to ULONGEST for
	reading values from registers.
---
 gdb/ChangeLog    | 5 +++++
 gdb/riscv-tdep.c | 2 +-
 2 files changed, 6 insertions(+), 1 deletion(-)

diff --git a/gdb/ChangeLog b/gdb/ChangeLog
index 4006dcbdf3..16bee04849 100644
--- a/gdb/ChangeLog
+++ b/gdb/ChangeLog
@@ -1,5 +1,10 @@
 2018-08-23  Andrew Burgess  <andrew.burgess@embecosm.com>
 
+	* riscv-tdep.c (riscv_return_value): Change type to ULONGEST for
+	reading values from registers.
+
+2018-03-06  Andrew Burgess  <andrew.burgess@embecosm.com>
+
 	* riscv-tdep.c (riscv_push_dummy_call): Use core_addr_to_string_nz
 	where appropriate.
 
diff --git a/gdb/riscv-tdep.c b/gdb/riscv-tdep.c
index 797ca72a84..3e1ad99c67 100644
--- a/gdb/riscv-tdep.c
+++ b/gdb/riscv-tdep.c
@@ -2318,7 +2318,7 @@ riscv_return_value (struct gdbarch  *gdbarch,
 	    /* Return value by reference will have its address in A0.  */
 	  case riscv_arg_info::location::by_ref:
 	    {
-	      CORE_ADDR addr;
+	      ULONGEST addr;
 
 	      regcache_cooked_read_unsigned (regcache, RISCV_A0_REGNUM,
 					     &addr);
-- 
2.18.0

